File: py/350. Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        r = []
        m = Counter(nums1)
        
        for n in nums2:
            if n in m and m[n] > 0:
                r.append(n)
                m[n] -= 1
        
        return r

File: cpp/350.IntersectionOfTwoArraysII.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        int arr[1001];
        vector <int> ans;
        fill_n(arr, 1001, 0);
        int sz1= nums1.size();
        int sz2= nums2.size();
        if(sz1>sz2)
        {
            for(int i=0; i<sz1; i++)
            arr[nums1[i]]++;
            for(int i=0; i<sz2; i++)
            {
                if(arr[nums2[i]]>0)
                {
                    ans.push_back(nums2[i]);
                    arr[nums2[i]]--;
                }
            }
        }
        else
        {
            for(int i=0; i<sz2; i++)
            arr[nums2[i]]++;
            for(int i=0; i<sz1; i++)
            {
                if(arr[nums1[i]]>0)
                {
                    ans.push_back(nums1[i]);
                    arr[nums1[i]]--;
                }
            }
        }
        return ans;
    }
};